### PR TITLE
Submission: py-dmgbuild (and py-ds-store, which the former depends on)

### PR DESCRIPTION
**Pull request by Lord-Kamina**
###### Type(s)

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

macOS 13.6.9
Xcode 15.2.0

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL in commit message? 
- [x] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?
- [ ] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

I have not included the dmgbuild tests as they do not work, but this is because the source archive appears to be missing files necessary for the tests to work properly.

### CODE REVIEW

File: python/py-dmgbuild/Portfile [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+checksums           md5     cc46f3a49ebbd014c83b6dd6faa16377 \
+                    rmd160  0a9436e0a4f98d95c031db32a5323f547533b4da \
+                    sha256  dba01d29f10c6804f2d72301600ddd03724daa41cd21a95409c44a3e199b19aa \
+                    size    36893
+
+patchfiles-append   patch-pyproject.patch
+
+python.versions     38 39 310 311 312
+python.pep517       yes
```
Comment (reneeotten) [inline python/py-dmgbuild/Portfile:28]: this is the default for Python 3.7+ so can be removed.

File: python/py-dmgbuild/Portfile [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+if {$subport ne $name} {
+    depends_lib     port:py${python.version}-mac-alias \
```
Comment (reneeotten) [inline python/py-dmgbuild/Portfile:31]: please use "append" as not to overwrite the pythonXY dependency added by the PortGroup

File: python/py-dmgbuild/Portfile [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+                    port:py${python.version}-ds-store
+
+    post-destroot {
+        xinstall -m 755 -d ${destroot}${prefix}/share/doc/${subport} \
+        ${destroot}${prefix}/share/examples/${subport}
+        xinstall -m 644 -W ${worksrcpath} LICENSE README.rst \
+        ${destroot}${prefix}/share/doc/${subport}
+    }
+    livecheck.type  none
```
Comment (reneeotten) [inline python/py-dmgbuild/Portfile:40]: this is already set by the "python" PG

File: python/py-ds-store/Portfile [hunk 4]
```diff
@@ -0,0 +1,39 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           python 1.0
+PortGroup           github 1.0
```
Comment (reneeotten) [inline python/py-ds-store/Portfile:5]: is this not available on PyPI - that is the preferred source unless there is a reason that doesn't work.

File: python/py-ds-store/Portfile [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+
+homepage            https://ds-store.readthedocs.io/en/latest/
+
+checksums           md5     2259e8cc693915eebf2aac1d43ebdd99 \
+                    rmd160  38d637cd6b1f203f75710026323a32141f99fac2 \
+                    sha256  00cf7d692829793a50e3f56660fcdda76988088ae319ad8db86330baeadded46 \
+                    size    24832
+
+python.versions     38 39 310 311 312
+python.pep517       yes
```
Comment (reneeotten) [inline python/py-ds-store/Portfile:27]: default already

File: python/py-ds-store/Portfile [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+if {$subport ne $name} {
+    depends_lib     port:py${python.version}-mac-alias
+
+    post-destroot {
+        xinstall -m 755 -d ${destroot}${prefix}/share/doc/${subport} \
+        ${destroot}${prefix}/share/examples/${subport}
+        xinstall -m 644 -W ${worksrcpath} LICENSE README.rst \
+        ${destroot}${prefix}/share/doc/${subport}
+    }
+    livecheck.type  none
```
Comment (reneeotten) [inline python/py-ds-store/Portfile:38]: set by the PortGroup

File: python/py-ds-store/Portfile [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+checksums           md5     2259e8cc693915eebf2aac1d43ebdd99 \
+                    rmd160  38d637cd6b1f203f75710026323a32141f99fac2 \
+                    sha256  00cf7d692829793a50e3f56660fcdda76988088ae319ad8db86330baeadded46 \
+                    size    24832
+
+python.versions     38 39 310 311 312
+python.pep517       yes
+
+if {$subport ne $name} {
+    depends_lib     port:py${python.version}-mac-alias
```
Comment (reneeotten) [inline python/py-ds-store/Portfile:30]: please use "-append"

File: python/py-ds-store/Portfile [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+long_description    ds_store lets you examine and modify .DS_Store files from Python code\; since it is written in pure Python, it is portable and will run on any platform, not just Mac OS X.
+
+homepage            https://ds-store.readthedocs.io/en/latest/
+
+checksums           md5     2259e8cc693915eebf2aac1d43ebdd99 \
+                    rmd160  38d637cd6b1f203f75710026323a32141f99fac2 \
+                    sha256  00cf7d692829793a50e3f56660fcdda76988088ae319ad8db86330baeadded46 \
+                    size    24832
+
+python.versions     38 39 310 311 312
```
Comment (reneeotten) [inline python/py-ds-store/Portfile:26]: can we only add recent Python versions?

File: python/py-dmgbuild/Portfile [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+homepage            https://dmgbuild.readthedocs.io/en/latest/
+
+checksums           md5     cc46f3a49ebbd014c83b6dd6faa16377 \
+                    rmd160  0a9436e0a4f98d95c031db32a5323f547533b4da \
+                    sha256  dba01d29f10c6804f2d72301600ddd03724daa41cd21a95409c44a3e199b19aa \
+                    size    36893
+
+patchfiles-append   patch-pyproject.patch
+
+python.versions     38 39 310 311 312
```
Comment (reneeotten) [inline python/py-dmgbuild/Portfile:27]: can we only add recent Python versions?

File: python/py-dmgbuild/Portfile [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+platforms           {darwin any}
+supported_archs     noarch
+maintainers         nomaintainer
+description         macOS command line utility to build disk images
+long_description \
+    dmgbuild is a command line tool to create macOS disk images (aka .dmg files). While it is possible to create disk images easily enough from the command line using the hdiutil program that ships with macOS, there is no easy way to configure the appearance of the resulting disk image when the user opens it.
+
+homepage            https://dmgbuild.readthedocs.io/en/latest/
+
+checksums           md5     cc46f3a49ebbd014c83b6dd6faa16377 \
```
Comment (reneeotten) [inline python/py-dmgbuild/Portfile:20]: "md5" is obsolete

File: python/py-ds-store/Portfile [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+license             {MIT PSF}
+platforms           {darwin any}
+supported_archs     noarch
+maintainers         nomaintainer
+description         Manipulate Finder .DS_Store files from Python
+long_description    ds_store lets you examine and modify .DS_Store files from Python code\; since it is written in pure Python, it is portable and will run on any platform, not just Mac OS X.
+
+homepage            https://ds-store.readthedocs.io/en/latest/
+
+checksums           md5     2259e8cc693915eebf2aac1d43ebdd99 \
```
Comment (reneeotten) [inline python/py-ds-store/Portfile:21]: "md5" is obsolete

**Review by reneeotten — CHANGES_REQUESTED** [review]

File: python/py-ds-store/Portfile (hunk 8 above)
Comment (Lord-Kamina) [reply python/py-ds-store/Portfile:26]: I'll change it to 3.9+.
I'm not usually a fan of bending over backwards to support dead software, but in this case it "just works"

File: python/py-dmgbuild/Portfile (hunk 10 above)
Comment (Lord-Kamina) [reply python/py-dmgbuild/Portfile:20]: Weird. I thought I did it with the value returned by `port` itself. Maybe I based myself off an old portfile or something like that?

File: python/py-ds-store/Portfile (hunk 4 above)
Comment (Lord-Kamina) [reply python/py-ds-store/Portfile:5]: No, it is. I think I just needed to set the distname appropriately.

**Review by reneeotten — APPROVED** [review]

**Comment by reneeotten** [comment]
thanks @Lord-Kamina !

**Pull request merged**